viewing_party/party.py

```diff
@@ -1,23 +1,141 @@
 # ------------- WAVE 1 --------------------
 
+from itertools import count
+
+
 def create_movie(title, genre, rating):
-    pass
+    if None in [title, genre, rating]:
+        return None
+
+    movie_dict = {
+                "title" : title,
+                "genre" : genre,
+                "rating" : rating
+    }
+    return movie_dict
+
+
+def add_to_watched(user_data, movie):
+    user_data["watched"].append(movie)
+    return user_data
+
+
+def add_to_watchlist(user_data, movie):
+    user_data["watchlist"].append(movie)
+    return user_data
+
+
+def watch_movie(user_data, title):
+    for i in range(len(user_data["watchlist"])):
+        if user_data["watchlist"][i]["title"] == title:
+            just_watched = user_data["watchlist"].pop(i)
+            user_data["watched"].append(just_watched) 
+    return user_data
+
 
 # -----------------------------------------
 # ------------- WAVE 2 --------------------
 # -----------------------------------------
+def get_watched_avg_rating(user_data):
+    sum = 0.00
+
+    if len(user_data["watched"]) == 0:
+        return sum

```

Comment:
This is more of a style preference, but I like to test for _falsy_ -ness in my guard clauses to  cover all falsy cases like:
- What if user_data is passed in as:
    - `None`
    - an empty string 
    - []

```python
if not user_data["watched"] or not user_data:
    return sum
```